### ISSUE TITLE
Provide a way to hide KVM CPUID from the guest

### ISSUE THREAD

**Issue by sboeuf** [cloud-hypervisor/cloud-hypervisor#2879]
Following the discussion https://github.com/cloud-hypervisor/cloud-hypervisor/discussions/2876, we might want to consider adding an option to let the user disable KVM specific CPUID leaves, so that the guest can't find out it's running inside a VM.
This is a workaround that is part of QEMU with the parameter `-cpu kvm=off`, and in practice is used to let NVIDIA gaming card being passed through. Without this workaround, the NVIDIA driver running in the guest realize it's running in a VM and refuses to load correctly, making the use of the GPU impossible.
If we want to implement this feature, we need to come up with a new parameter (something like `hide_kvm=on|off`), with the option being `off` by default. When enabled we would clear the CPUID leaves related to KVM, that is:
- 0x40000000 `KVM_CPUID_SIGNATURE`
- 0x40000001 `KVM_CPUID_FEATURES`

**Comment by iggy** [comment]
Not that I don't think this idea is a good one, but recent versions of the Nvidia driver load fine without workarounds anymore.

I think a more generic option to hide/show various bits of cpuid would be a better long term solution.

**Comment by rbradford** [comment]
This is a bad idea. This bit is used by Linux to determine use of PV features.

**Comment by weltling** [comment]
I guess also there might be issues with the Windows guest. At least from what I saw, many code paths depend on whether a hypervisor is present. E.g. recall the timer source investigation where we ended up using the kvm enlightenments. This might be still doable, but putting this to just be aware. E.g. one would still go for implementing HPET if it regards timer or see what other hardware would be affected and come in question. Also, perhaps NVIDIA would behave a different way under under Windows, would need an additional check. 

Thanks
